src/VisionModel/resnet.jl

```diff
@@ -0,0 +1,86 @@
+using Flux
+
+export ResNet18, ResNet34, ResNet50, ResNet101, ResNet152
+
+basicblock(inplanes, outchannels, downsample = false) = downsample ? 
+  Chain(Conv((3, 3), inplanes => outchannels[1], stride = 2),
+        BatchNorm(outchannels[1], λ = relu),
+        Conv((3, 3), outchannels[1] => outchannels[2], stride = 1, pad = 1),
+        BatchNorm(outchannels[2], λ=relu)) : 
+  Chain(Conv((3, 3), inplanes => outchannels[1], stride = 1),
+        BatchNorm(outchannels[1], λ = relu),
+        Conv((3, 3), outchannels[1] => outchannels[2], stride = 1, pad = 1),
+        BatchNorm(outchannels[2], λ=relu))
+
+bottleneck(inplanes, outchannels, downsample = false) = downsample ?
+  Chain(Conv((1, 1), inplanes => outchannels[1], stride = 2),
+        BatchNorm(outchannels[1], λ = relu),
+        Conv((3, 3), outchannels[1] => outchannels[2], stride = 1, pad = 1),
+        BatchNorm(outchannels[2], λ = relu),
+        Conv((1, 1), outchannels[2] => outchannels[3], stride = 1),
+        BatchNorm(outchannels[3], λ = relu)) :
+  Chain(Conv((1, 1), inplanes => outchannels[1], stride = 1),
+        BatchNorm(outchannels[1], λ = relu),
+        Conv((3, 3), outchannels[1] => outchannels[2], stride = 1, pad = 1),
+        BatchNorm(outchannels[2], λ = relu),
+        Conv((1, 1), outchannels[2] => outchannels[3], stride = 1),
+        BatchNorm(outchannels[3], λ = relu))
+
+projection(inplanes, outplanes, stride) = Chain(Conv((1, 1), inplanes => outplanes, stride=stride),
+                                                               BatchNorm((outplanes), λ=relu))
+# array -> PaddedView(0, array, outplanes) for zero padding arrays
+identity(inplanes, outplanes, stride) = inplanes < outplanes ? 
+ array ->  cat(array, zeros(Float32, outplanes - inplanes); dims = (1, 2, 3)) : +
```

Comment:
Instead of explicitly specifying `Float32` here, we can do `eltype(array)` to respect the element type of the array passed in.